rust/sbd-server/src/lib.rs

```diff
@@ -233,11 +233,78 @@ impl SbdServer {
         // read this as a prioritization of existing connections over incoming
         let connect_limit = Arc::new(tokio::sync::Semaphore::new(1024));
 
-        let weak_cslot = cslot.weak();
+        let mut bind_port_zero = Vec::new();
+        let mut bind_explicit_port = Vec::new();
+
         for bind in config.bind.iter() {
             let a: std::net::SocketAddr = bind.parse().map_err(Error::other)?;
 
-            let tcp = tokio::net::TcpListener::bind(a).await?;
+            if a.port() == 0 {
+                bind_port_zero.push(a);
+            } else {
+                bind_explicit_port.push(a);
+            }
+        }
+
+        let (mut listeners, mut l2) = tokio::join!(
+            async {
```

Comment:
So this block will try to bind all the interfaces that are specifying port 0, on the same port. But if that fails, then let them all bind on different ports? So this should never fail but will try to be tidy. That seems neat!

Maybe a doc comment to explain the intention though, because the inline comments are quite enough for me to be able to read this off without looking through the code.

Reply:
How about in the cli `--help`?: https://github.com/holochain/sbd/pull/22/commits/4d57ac05e62f3063404a480c7109863c1db7f191